packages/shared_preferences/shared_preferences_android/android/src/test/kotlin/io/flutter/plugins/sharedpreferences/SharedPreferencesTest.kt

```diff
@@ -217,7 +236,26 @@ internal class SharedPreferencesTest {
   fun testSetAndGetStringListWithSharedPreferences() {
     val plugin = pluginSetup(sharedPreferencesOptions)
     plugin.setEncodedStringList(listKey, testList, sharedPreferencesOptions)
-    Assert.assertEquals(plugin.getStringList(listKey, sharedPreferencesOptions), testList)
+    val result = plugin.getStringList(listKey, sharedPreferencesOptions)
+    Assert.assertEquals(result?.jsonEncodedValue, testList)
+  }
+
+  @Test
+  fun testSetAndGetStringListWithSharedPreferencesRedirectsForLegacy() {
```

Comment:
non blocking nit: would prefer "...DeprecatedStringList" over "Legacy" but this is a test so the consequence is small.

Reply:
Renamed to the more specific "PlatformEncoded".